raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -514,6 +514,16 @@ public UnattachedState unattachedStateOrThrow() {
         throw new IllegalStateException("Expected to be Leader, but current state is " + state);
     }
 
+    @SuppressWarnings("unchecked")
+    public <T> Optional<LeaderState<T>> maybeLeaderState() {
+        EpochState state = this.state;
+        if (state instanceof  LeaderState) {
+            return Optional.of((LeaderState<T>) state);
+        } else {
+            return Optional.empty();
+        }
+    }
```

Comment:
Thanks you. You read my mind. I was thinking of something similar after I left my previous comment.